### PR TITLE
Add terser (JS compressor for ES5 and ES6)

### PR DESCRIPTION
**Pull request by felix-last**
Add [terser](https://github.com/terser/terser) as a JS compressor which can handle both ES5 and ES6 syntax.

### CODE REVIEW

**Review by mathjazz — COMMENTED** [review]
This looks good, except for the conflict.

~Could you please rebase against latest master?~

UPDATE: Now that I have access, I could do it myself. :)

**Comment by jezdez** [comment]
@mathjazz Seems like the tests failed for CI related issues in lastest Django master. I've fixed it in master..

**Comment by mathjazz** [comment]
Thanks! Master seems to be failing, though.

**Comment by jezdez** [comment]
> Thanks! Master seems to be failing, though.

Yeah, seems like some things that got removed following the deprecation policy. I've just opened #742 to be clear what's failing.

**Review by mathjazz — APPROVED** [review]

File: tests/tests/test_compressor.py [hunk 1]
```diff
@@ -240,7 +240,12 @@ def test_csshtmljsminify(self):
     def test_uglifyjs(self):
         self._test_compressor('pipeline.compressors.uglifyjs.UglifyJSCompressor',
             'js', 'pipeline/compressors/uglifyjs.js')
-        
+
+    @skipUnless(settings.HAS_NODE, "requires node")
```
Comment (jezdez) [inline tests/tests/test_compressor.py:244]: @mathjazz This will need an update to the `package.json` so terser is actually installed for the tests.

**Review by jezdez — CHANGES_REQUESTED** [review]

File: tests/tests/test_compressor.py (hunk 1 above)
Comment (mathjazz) [reply tests/tests/test_compressor.py:244]: I thought https://github.com/jazzband/django-pipeline/pull/696/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519 does that?

Comment (jezdez) [reply tests/tests/test_compressor.py:244]: You're right, I'm blind :)

**Comment by mathjazz** [comment]
Hey @cyberdelia, we'd like to switch https://github.com/mozilla/pontoon/pull/1819#issuecomment-762189263 to the terser compressor, support for which just landed on master.

Could you please publish the new release on PyPI?

**Comment by cyberdelia** [comment]
just created the 2.0.6 release.
Thanks a lot @jezdez for making it so easy.

**Comment by jezdez** [comment]
> just created the 2.0.6 release.
> Thanks a lot @jezdez for making it so easy.

Happy you feel that way! Remember to release the files to PyPI on https://jazzband.co/projects/django-pipeline#2.0.6

**Comment by julen** [comment]
@cyberdelia any chance to have the 2.0.6 package uploaded to PyPI? 🙏 Thanks!

**Comment by cyberdelia** [comment]
Done.

**Pull request merged**